src/web100srv.c

```diff
@@ -1434,11 +1315,16 @@ int run_test(tcp_stat_agent* agent, int ctlsockfd, TestOptions* testopt,
              vars.Timeouts, vars.Sndbuf, vars.MaxRwinRcvd, vars.CurrentCwnd,
              mismatch, bad_cable, totalcnt, refresh);
   }
-  shutdown(ctlsockfd, SHUT_WR);
+  if (ctl->ssl != NULL) {
+    SSL_shutdown(ctl->ssl);
+  } else {
+    shutdown(ctl->socket, SHUT_WR);
+  }
```

Comment:
I mean that I have changed it to use close_connection as part of the cleanup you asked for in a previous comment.